### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by AShivangi**
- ios -> ios_base: seems like ios_flags_restore class wasn't really fixing the coverity issue. Not sure if this change will help

- getenv needs to be sanitized before being used. Coverity requires an annotation to be added for custom sanitization functions. This change should fix that issue

- fixing a couple of other coverity issues as well

### CODE REVIEW

**Comment by AShivangi** [comment]
retest this please

**Comment by abalasa** [comment]
retest this please

File: src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp [hunk 1]
```diff
@@ -231,12 +231,12 @@ XQSPIPS_Flasher::XQSPIPS_Flasher(std::shared_ptr<pcidev::pci_device> dev)
     // By default, it is 'perallel'
     mConnectMode = 0;
     // U30 Rev.A use single
-    if (typeStr.find("single")) {
+    if (typeStr.find("single") != std::string::npos) {
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp]: shouldn't this needs to be if (typeStr.find("single") != 0)?

File: src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp [hunk 2]
```diff
@@ -231,12 +231,12 @@ XQSPIPS_Flasher::XQSPIPS_Flasher(std::shared_ptr<pcidev::pci_device> dev)
...
         mConnectMode = 1;
     }
 
     mBusWidth = 4;
-    if (typeStr.find("x2")) {
+    if (typeStr.find("x2") != std::string::npos) {
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp]: ditto.

File: src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp]: I thought that's how "find" is used. Search for "single" until the end of the string

Comment (maxzhen) [reply src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp]: But std::string::npos is -1, how does it match the original code here? @mamin506 , please confirm.

Comment (mamin506) [reply src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp]: typeStr.find("single") != std::string::npos is correct. If no matched was found, find() returns string::npos. The original code will failed if 'single' is the start of the string.

File: src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp (hunk 2 above)
Comment (mamin506) [reply src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp]: See above

**Pull request merged**